x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -1507,9 +1512,7 @@ export class Embeddable
     return this.viewUnderlyingDataArgs;
   }
 
-  public canViewUnderlyingData() {
-    return this.loadViewUnderlyingDataArgs();
-  }
+  public canViewUnderlyingData$ = new BehaviorSubject<boolean>(false);
```

Comment:
I needed to turn this into an observable to implement `couldBecomeCompatible` and `subscribeToCompatibilityChanges` because the `Open in Discover` action wasn't showing in the hover actions when you open a dashboard because it doesn't meet the compatibility checks on initial load, and so the action would become compatible after the initial load. The hover actions would then become out of sync with the available actions and missing the `Open in Discover` action because it wasn't triggering a re-render. (see https://github.com/elastic/kibana/pull/182535/files#diff-3eb916180bfc6eb3a79a7be408ce36dc05b9fdbdb7e6e35fc1f10c2f20729ea2R51-R65)